src/containers/hashlog/Hashlog.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import { Container, Row, Col } from 'reactstrap';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import { getBlockListPage, setBlockListPage, setBlockListPageSize } from '../../actions/hashlog';
+import ReactTable from '../../components/SelectableReactTable';
+import createMqProvider, {querySchema} from '../../MediaQuery';
+import PaginationWithPage from '../../components/PaginationWithPage';
+
+const {MediaQuery, Screen} = createMqProvider(querySchema);
+
+class Hashlog extends React.Component {
+  state = {
+    selectedRound: 0,
+    nameFilter: '',
+    sort: {},
+  };
+
+  componentDidMount() {
+    const {getBlocksPages, blocksPage, blocksPageSize} = this.props;
+    getBlocksPages(blocksPage, blocksPageSize);
+  }
+
+  renderBlocklist() {
+    const {
+      hashlog: {blockList, totalBlocks, blocksPage, blocksPageSize},
+      setBlocksPage,
+      setBlocksPageSize,
+      getBlocksPages
+    } = this.props;
+    return (
+      <React.Fragment>
+        <MediaQuery>
```

Comment:
MediaQuery should be mount once for whole app at the root and the Screen component should be mounted where it's needed.